File: 001. Two Sum/main.go
// package main

// import (
// 	"fmt"
// )

func twoSum(nums []int, target int) []int {
	var result []int

	numsLen := len(nums)
	for i := 0; i < numsLen; i++ {
		for n := (i + 1); n < numsLen; n++ {
			if (nums[i] + nums[n]) == target {
				result = append(result, i, n)
				return result
			}
		}
	}
	return nil
}

// func main() {
// 	nums := []int{3, 2, 4}
// 	var target = 6

// 	fmt.Println(twoSum(nums, target))
// }

File: 001. Two Sum/main.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */

int* twoSum(int* nums, int numsSize, int target, int* returnSize){
    int i = 0;
    int n = 0;
    int *result;
    result = malloc(2 * sizeof(int));
    while (i < numsSize) {
        n = i + 1;
        while (n < numsSize) {
            if ((nums[i] + nums[n]) == target) {
                result[0] = i;
                result[1] = n;
                *returnSize = 2;
                return (result);
            }
            n++;
        }
        i++;
    }
    *returnSize = 0;
    return (0);
}
